app/schemas/models/user.coffee

```diff
@@ -417,6 +418,36 @@ _.extend UserSchema.properties,
         additionalProperties: FeatureRecipientSchema
 
   archived: c.date {description: 'Marks this record for automatic online archiving to cold storage by our cloud database.'}
+  products: c.array {title: 'Products purchased or used by this user'},
+    c.object { required: ['product', 'startDate', 'recipient', 'paymentService', 'paymentDetails'],  additionalProperties: true },
+      # ensure we can add additionalProperties
+      product: { type: 'string', enum: ['course', 'basic_subscription', 'pd', 'ai-league', 'online-classes'], decription: 'The "name" field for the product purchased' }  # And/or the ID of the Product in the database, if we make a Product for each thing we can buy?
+
+      prepaid: c.objectId(links: [ {rel: 'db', href: '/db/prepaid/{($)}'} ])  # required for type: “course” for legacy compatibility, optional for other types
```

Comment:
doesn't make sense to call it `prepaid` if we are going to keep it at top-level in products. Maybe uid or something so that we can add other ids here for logging/querying

Reply:
this is not a `required` key :joy: so let's make it in top-level 😂  we can add other key name (for example productId) to refer to other products. i wanted to call it productId but that's confusing for prepaid (not a product)... and this key connect to lots of logic so i don't refactor it anymore 😂